src/test/java/com/artipie/asto/etcd/EtcdStorageITCase.java

```diff
@@ -5,54 +5,105 @@
 
 package com.artipie.asto.etcd;
 
+import com.artipie.asto.Content;
 import com.artipie.asto.Key;
 import com.artipie.asto.Storage;
 import com.artipie.asto.blocking.BlockingStorage;
+import com.github.dockerjava.api.DockerClient;
 import io.etcd.jetcd.Client;
-import io.etcd.jetcd.launcher.EtcdCluster;
+import io.etcd.jetcd.launcher.EtcdContainer;
 import io.etcd.jetcd.test.EtcdClusterExtension;
+import java.nio.charset.StandardCharsets;
 import java.util.concurrent.CompletionException;
 import org.hamcrest.MatcherAssert;
 import org.hamcrest.Matchers;
 import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.DockerClientFactory;
 
 /**
  * Test case for etcd-storage.
  * @since 1.0
- * @todo #306:90min The test is disabled,
- *  the CI can't start ETCD extension which depends on testcontainer with
- *  etcd Docker image. Let's fix this issue on CI and enable the test.
  * @todo #306:90min Add etcd storage to StorageExtension to run all common
  *  tests on EtcdStorage too. It could be not an easy task, since etcd
  *  test depends on etcd clust with at least one node. For this test
  *  it starts using testcontainers and `EtcdCluster` junit extension.
  */
-@Disabled
-public final class EtcdStorageITCase {
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+final class EtcdStorageITCase {
 
     /**
      * Test cluster.
      */
     @RegisterExtension
-    static final EtcdCluster ETCD = new EtcdClusterExtension("test-etcd", 1);
+    static final EtcdClusterExtension ETCD = new EtcdClusterExtension(
+        "test-etcd",
+        1,
+        false,
+        "--data-dir",
+        "/data.etcd0"
+    );
 
     /**
      * Storage.
      */
     private Storage storage;
 
+    @BeforeAll
+    static void beforeAll() throws InterruptedException {
+        final DockerClient client = DockerClientFactory.instance().client();
+        client.pullImageCmd(EtcdContainer.ETCD_DOCKER_IMAGE_NAME)
+            .start()
+            .awaitCompletion();
+        ETCD.start();
+    }
+
     @BeforeEach
     void setUp() {
         this.storage = new EtcdStorage(
             Client.builder().endpoints(ETCD.getClientEndpoints()).build()
         );
     }
 
+    @AfterAll
+    static void afterAll() {
+        ETCD.close();
```

Comment:
@baudoliver7 we also need to stop docker conteiber here if we start it in beforeall method

Reply:
@g4s8 This is exactly what we do with the instruction:
```java
ETCD.close();
```
This instruction close the cluster :
```java
public void close() {
    this.cluster.close();
}
```
And the cluster close all containers :
```java
public void close() {
    containers.forEach(EtcdContainer::close);
}
```
This is the code of method `EtcdContainer::close` :
```java
public void close() {
    if (this.container != null) {
        this.container.stop();
    }

    deleteDataDirectory(this.dataDirectory);
}
```